src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/ComplexFlowValidationResponse.groovy

```diff
@@ -0,0 +1,50 @@
+package org.openkilda.functionaltests.helpers.model
+
+import org.openkilda.northbound.dto.v1.flows.FlowValidationDto
+import org.openkilda.northbound.dto.v1.flows.PathDiscrepancyDto
+import org.openkilda.northbound.dto.v2.haflows.HaFlowValidationResult
+import org.openkilda.northbound.dto.v2.yflows.YFlowValidationResult
+
+import groovy.transform.ToString
+
+@ToString
+class ComplexFlowValidationResponse {
+    private static final String COOKIE_ID_IN_RULE_DISCREPANCY_STRING_REGEX =  /-?\d{19}/
+
+    boolean asExpected
+    List<SubFlowValidationDiscrepancies> subFlowsDiscrepancies
+
+    ComplexFlowValidationResponse(YFlowValidationResult yFlowValidation) {
+        this.asExpected = yFlowValidation.asExpected
+        this.subFlowsDiscrepancies = collectDiscrepanciesPerSubFlow(yFlowValidation.subFlowValidationResults)
+    }
+
+    ComplexFlowValidationResponse(HaFlowValidationResult haFlowValidation) {
+        this.asExpected = haFlowValidation.asExpected
+        this.subFlowsDiscrepancies = collectDiscrepanciesPerSubFlow(haFlowValidation.subFlowValidationResults)
+
+    }
+
+    static private List<SubFlowValidationDiscrepancies> collectDiscrepanciesPerSubFlow(List<FlowValidationDto> subFlowValidationDetails) {
```

Comment:
it might be better to use "private static" instead of "static private". 
for the sake of common approach